judgels-frontends/raphael/src/components/ProblemWorksheetCard/Bundle/ProblemStatementCard/ItemEssayCard/ItemEssayForm/ItemEssayForm.test.tsx

```diff
@@ -0,0 +1,236 @@
+          cancelButtonState: AnswerState.NotAnswered,
+        };
+        expect(state).toEqual(expectedState);
+      });
+
+      it('textarea', () => {
+        const textareaValue = wrapper.find('textarea').props().value;
+        expect(textareaValue).toEqual('');
+      });
+
+      it('only answer button', () => {
+        const button = wrapper.find('button');
+        expect(button.length).toEqual(1);
+        expect(button.text()).toEqual('Answer');
+      });
+
+      it('helptext', () => {
+        const helpText = wrapper
+          .find('div')
+          .last()
+          .text();
+        expect(helpText).toContain('Not answered.');
+      });
+    });
+
+    describe('fill and submit the answer', () => {
+      beforeEach(async () => {
+        const answerButton = wrapper.find('button').first();
+        await answerButton.simulate('submit');
+        const textarea = wrapper.find('textarea').first();
+        textarea.simulate('change', { target: { value: 'answer' } });
+      });
+
+      it('fill the answer', () => {
+        const state = wrapper.state();
+        expect(state.answer).toEqual('answer');
+        expect(state.answerState).toEqual(AnswerState.Answering);
+      });
+
+      it('cancel answer', async () => {
+        const cancelButton = wrapper.find('button').last();
+        await cancelButton.simulate('click');
+        const state = wrapper.state();
+        const expectedState = {
+          answerState: AnswerState.NotAnswered,
+          answer: '',
+          initialAnswer: '',
+          cancelButtonState: AnswerState.NotAnswered,
+        };
+        expect(state).toEqual(expectedState);
+      });
+
+      it('submit the answer', async () => {
+        const submitButton = wrapper.find('button').first();
+        await submitButton.simulate('submit');
+        const state = wrapper.state();
+        const expectedState = {
+          answerState: AnswerState.AnswerSaved,
+          answer: 'answer',
+          initialAnswer: 'answer',
+          cancelButtonState: AnswerState.AnswerSaved,
+        };
+        expect(state).toEqual(expectedState);
+      });
+    });
+  });
+
+  describe('answered', () => {
+    const props: ItemEssayFormProps = {
+      jid: 'jid',
+      type: ItemType.Essay,
+      meta: 'meta',
+      config: itemConfig,
+      disabled: false,
+      initialAnswer: 'initial',
+      onSubmit: jest.fn().mockReturnValue(undefined),
+      answerState: AnswerState.AnswerSaved,
+    };
+
+    beforeEach(() => {
+      wrapper = mount(<ItemEssayForm {...props} />);
+    });
+
+    describe('initial condition', () => {
+      it('initial state', () => {
+        const state = wrapper.state();
+        const expectedState = {
+          answerState: props.answerState,
+          answer: props.initialAnswer,
+          initialAnswer: props.initialAnswer,
+          cancelButtonState: AnswerState.AnswerSaved,
+        };
+        expect(state).toEqual(expectedState);
```

Comment:
Please do not test state. State is a component's implementation details.

See: https://kentcdodds.com/blog/testing-implementation-details